ambry-store/src/main/java/com.github.ambry.store/LogSegment.java

```diff
@@ -180,6 +182,37 @@ public void appendFrom(ReadableByteChannel channel, long size) throws IOExceptio
     }
   }
 
+  /**
+   * {@inheritDoc}
+   * <p/>
+   * Attempts to write the {@code byteArray} to this segment in direct IO manner.
+   * <p/>
+   * The write is not started if it cannot be completed.
+   * @param byteArray The bytes from which data needs to be written from.
+   * @param offset The offset in the byteArray to start with.
+   * @param length The amount of data in bytes to use from the byteArray.
+   * @throws IllegalArgumentException if there is not enough space for data of size {@code length}.
+   * @throws IOException if data could not be written to the file because of I/O errors
+   */
+  public int appendFromDirectly(byte[] byteArray, int offset, int length) throws IOException {
+    if (!fileChannel.isOpen()) {
+      throw new ClosedChannelException();
+    }
+    if (endOffset.get() + length > capacityInBytes) {
```

Comment:
maybe we can have a function for this since this is shared by two functions?

Reply:
Fixed.